pkg/runtime/runtime.go

```diff
@@ -585,6 +588,14 @@ func (a *DaprRuntime) beginPubSub(name string, ps pubsub.PubSub) error {
 				log.Debugf("no matching route for event %v in pubsub %s and topic %s; skipping", cloudEvent[pubsub.IDField], name, msg.Topic)
 				return nil
 			}
+			if dataAsPayload && !rawPayload {
+				cloudData := cloudEvent[pubsub.DataField]
```

Comment:
I was just thinking of this feature and noticed this PR , data can be in two different fields in the CloudEvent so this logic would suffice. I suggest taking a look at the method `publishMessageGRPC` and maybe even to extract to a new method the log get the `data` from the cloudEvent

Reply:
I don't think `GRPC` needs to do so, it already has a data field. Just as the related issue, this is only for HTTP.